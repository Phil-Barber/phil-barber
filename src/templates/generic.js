import React from 'react';
import styled from 'styled-components';
import PageWrapper from '../components/pageWrapper';
import SEO from '../components/seo';
import { graphql } from 'gatsby';

const Container = styled.div`
  max-width: 700px;
  text-align: justify;
  margin: 0 auto;
`;

export const Generic = ({ data }) => (
  <PageWrapper>
    <SEO
      title={data.markdownRemark.frontmatter.title}
      description={data.markdownRemark.export}
    />
    <Container>
      <h1>{data.markdownRemark.frontmatter.title}</h1>
      <div dangerouslySetInnerHTML={{ __html: data.markdownRemark.html }} />
    </Container>
  </PageWrapper>
);

export default Generic;

export const query = graphql`
  query PostQuery($slug: String!) {
    markdownRemark(fields: { slug: { eq: $slug } }) {
      html
      excerpt
      frontmatter {
        title
      }
    }
  }
`;
