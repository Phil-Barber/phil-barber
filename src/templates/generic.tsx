import * as React from 'react';
import styled from 'styled-components';
import { FullWidthWrapper } from '../components/pageWrapper';
import SEO from '../components/seo';
import { graphql } from 'gatsby';
import { MarkdownRemark } from '../types/graphql-types';

const Container = styled.div`
  max-width: 700px;
  text-align: justify;
  margin: 0 auto;
`;

interface Props {
  readonly data: {
    readonly markdownRemark: MarkdownRemark;
  };
}

export const Generic: React.FC<Props> = ({ data }: Props) => (
  <FullWidthWrapper>
    <SEO
      title={data.markdownRemark.frontmatter.title}
      description={data.markdownRemark.excerpt}
    />
    <Container>
      <h1>{data.markdownRemark.frontmatter.title}</h1>
      <div dangerouslySetInnerHTML={{ __html: data.markdownRemark.html }} />
    </Container>
  </FullWidthWrapper>
);

export default Generic;

export const query = graphql`
  query PostQuery($slug: String!) {
    markdownRemark(fields: { slug: { eq: $slug } }) {
      html
      excerpt
      frontmatter {
        title
      }
    }
  }
`;
