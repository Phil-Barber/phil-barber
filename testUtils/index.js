import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../src/style';
import '@testing-library/jest-dom/extend-expect';

const customRender = (ui, options, store) => {
  const AllTheProviders = ({ children }) => (
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  );
  return render(ui, { wrapper: AllTheProviders, ...options });
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
