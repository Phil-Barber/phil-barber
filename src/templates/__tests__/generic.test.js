import React from 'react';
import { render } from '../../../testUtils';
import { Generic } from '../generic';

describe('Generic', () => {
  it('Matches snapshot', () => {
    const { container } = render(
      <Generic
        data={{
          markdownRemark: {
            html: '<div>Test Page</div>',
            excerpt: 'Test excerpt',
            frontmatter: {
              title: 'Title',
            },
          },
        }}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
