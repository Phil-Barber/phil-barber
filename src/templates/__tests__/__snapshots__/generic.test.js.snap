// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generic Matches snapshot 1`] = `
<div>
  <div
    class="headroom-wrapper"
  >
    <div
      class="headroom headroom--unfixed"
      style="position: relative; top: 0px; left: 0px; right: 0px; z-index: 1; transform: translate3D(0, 0, 0);"
    >
      <div
        class="sc-AxjAm fEzfgO"
      >
        <a
          class="sc-AxhUy bILkkK"
          href="/"
        >
          <h1>
            Test title
          </h1>
        </a>
        <span
          class="sc-AxirZ gtVPNy"
          style="transform: rotate(0deg);"
        >
          &gt;
        </span>
      </div>
    </div>
  </div>
  <div
    class="sc-AxiKw jmqPVm sc-Axmtr eCLTHA"
  >
    <div
      class="sc-AxheI hhwjIJ"
    >
      <div
        class="sc-AxmLO bIlwlq"
      >
        <h1>
          Title
        </h1>
        <div>
          <div>
            Test Page
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-AxhCb bdrWTY"
    />
  </div>
</div>
`;
